### ISSUE TITLE
Return type of image depends on settings

### ISSUE THREAD

**Issue by Johannes-HAWHH** [mmackh/IPDFCameraViewController#31]
xCode: 8.0 (8A218a)
Lang: Swift 3
IPDFCameraViewController: Latest version as of this post (Podfile.lock says 0.0.1)

When calling the captureImage function the return type of the "image" variable depends on the setup you did before. Usually you get image as a UIImage which is completely fine. However if neither IPDFCameraViewTypeBlackAndWhite nor isBorderDetectionEnabled are true the return type of "image" will be Data instead of UIImage which is an unexpected behavior.

I don't know if this is the intended behavior but I didn't find anything about that in the readme so I guess it's a bug.

This is a workaround I'm using right now:
```swift
@IBAction func captureButtonTapped(_ sender: AnyObject) {
        let imageCaptureCompletionHandler: ((Any?) -> Void) = { [weak self] image in
            if let unwrappedSelf = self {
                
                // Case: Image provided as UIImage.
                if let unwrappedUIImage = image as? UIImage {
                    unwrappedSelf.selectedImage = unwrappedUIImage
                }
                
                // Case: Image provided as Data.
                if let unwrappedImageData = image as? Data {
                    unwrappedSelf.selectedImage = UIImage(data: unwrappedImageData)
                }
                
                unwrappedSelf.showImagePreviewView()
            }
        }
        
        cameraViewController.captureImage(completionHander: imageCaptureCompletionHandler)
    }
```

If this should be the intended behavior it should probably be stated somewhere.

Anyways big thank you for this framework - nice work!

**Comment by GoSwiftlyIntoThisGoodNight** [comment]
Yeah, this is a pretty big problem. Why is it not just UIImage always?

**Comment by mmackh** [comment]
I think this maybe an issue with CocoaPods. I don't use it at all and the version may be off. I'm returning NSString filePaths

**Comment by JayJayDF** [comment]
I've looked into the code and the problem simply is that the cast happens inside the condition block which makes sure that either IPDFCameraViewTypeBlackAndWhite or isBorderDetectionEnabled are true. You could move the cast outside of the condition block or just add a Data to UIImage cast inside the else part which is executed if none of those settings are true.

However I feel like the CocoaPods version is pretty outdated anyways. If you're "not using it at all" why do you offer it then in the readme:

> Cocoapods
> 
> Add the following to your Podfile:
> 
> use_frameworks!
> pod "IPDFCameraViewController"


I would really appreciate if you'd update the pods version. If for some reason you don't wanna do that I think you should remove it from pods entirely because atm it's very misleading.

P.S.: This is the part in question (relevant lines only):

```
- (void)captureImageWithCompletionHander:(void(^)(id data))completionHandler
{
         [...]
         if (weakSelf.cameraViewType == IPDFCameraViewTypeBlackAndWhite || weakSelf.isBorderDetectionEnabled)
         {
                 [...]
                 UIImage *image = UIGraphicsGetImageFromCurrentImageContext();
                 [...]
                 completionHandler(image);
         }
         else
         {
                  [...]
                  completionHandler(imageData);
         }
}
```
